lib/internal/buffer.js

```diff
@@ -56,6 +56,77 @@ function boundsError(value, length, type) {
 }
 
 // Read integers.
+function readBigUInt64LE(offset) {
+  checkNumberType(offset);
+  const first = this[offset];
+  const last = this[offset + 7];
+  if (first === undefined || last === undefined)
+    boundsError(offset, this.length - 8);
+
+  // TODO: do we need tricky math from readUInt48LE?
```

Comment:
You mean the `(a + b * 256) * 2 ** 32` bit?

I expect it's faster to compute the high and low words as 32 bits quantities first, then convert them to bigints:
```js
const lo = /* ... */;
const hi = /* ... */;
return BigInt(lo) + BigInt(hi) * 2n ** 32n;
```
I'm unsure if V8 currently constant-folds the `2n ** 32n` so you might want to cache that.

Reply:
It's probably faster now, but I would prefer to put off any optimizations until V8 6.7 is out.

Reply:
Have you considered using `A << Bn` instead of `A * 2n ** Bn` (everywhere)?
It's shorter and IMO easier to read.

Reply:
it's also much more easily optimized :D

Reply:
I chose to use the same approach as in other read functions.

Reply:
Caching is indeed faster:

```
                                                                    confidence improvement accuracy (*)    (**)   (***)
 buffers\\buffer-read.js n=1000000 type='BigInt64BE' buffer='fast'         ***     36.14 %      Â±4.71% Â±6.26% Â±8.15%
 buffers\\buffer-read.js n=1000000 type='BigInt64BE' buffer='slow'         ***     32.26 %      Â±5.68% Â±7.58% Â±9.92%
 buffers\\buffer-read.js n=1000000 type='BigInt64LE' buffer='fast'         ***     37.72 %      Â±4.18% Â±5.58% Â±7.29%
 buffers\\buffer-read.js n=1000000 type='BigInt64LE' buffer='slow'         ***     34.75 %      Â±5.00% Â±6.66% Â±8.67%
 buffers\\buffer-read.js n=1000000 type='BigUInt64BE' buffer='fast'        ***     34.65 %      Â±3.41% Â±4.56% Â±5.97%
 buffers\\buffer-read.js n=1000000 type='BigUInt64BE' buffer='slow'        ***     35.73 %      Â±3.70% Â±4.92% Â±6.42%
 buffers\\buffer-read.js n=1000000 type='BigUInt64LE' buffer='fast'        ***     30.61 %      Â±3.91% Â±5.20% Â±6.78%
 buffers\\buffer-read.js n=1000000 type='BigUInt64LE' buffer='slow'        ***     32.13 %      Â±3.36% Â±4.47% Â±5.83%
```

But using bitwise shift instead of multiplication as @targos suggested is equally fast, and less ugly than a constant, so I went with that.